### PR TITLE
Add ListField example to apply token indexers

### PR DESCRIPTION
**Pull request by JohnGiorgi**
Addresses a [comment in the migration guide](https://github.com/allenai/allennlp/discussions/4933#discussioncomment-371349).

Changes proposed in this pull request:

- Added an example for fields of type `ListField[TextField]` to `apply_token_indexers` API docs.

The documentation for `apply_token_indexers` now reads like this:

image

Let me know if there's anything you would like me to change! @epwalsh

### CODE REVIEW

File: allennlp/data/dataset_readers/dataset_reader.py [hunk 1]
```diff
@@ -233,6 +233,14 @@ def apply_token_indexers(self, instance: Instance) -> None:
         """
         If `Instance`s created by this reader contain `TextField`s without `token_indexers`,
         this method can be overriden to set the `token_indexers` of those fields.
+
```
Comment (epwalsh) [inline allennlp/data/dataset_readers/dataset_reader.py:236]:
```suggestion

        E.g. if you have you have `"source"` `TextField`, you could implement this method like this:

        ```python
        def apply_token_indexers(self, instance: Instance) -> None:
            instance["source"].token_indexers = self._token_indexers
        ```

```

**Review by epwalsh — COMMENTED** [review]
Thanks @JohnGiorgi! I just left a suggestion to add an example of the case where you have `TextField` by itself.

**Comment by JohnGiorgi** [comment]
> Thanks @JohnGiorgi! I just left a suggestion to add an example of the case where you have `TextField` by itself.

Good call! Committed your suggestion

**Review by epwalsh — APPROVED** [review]
LGTM!

**Pull request merged**